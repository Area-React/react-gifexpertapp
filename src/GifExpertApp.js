import React, { useState } from 'react';
import { AddCategory } from './tests/components/AddCategory';
import { GifGrid } from './tests/components/GifGrid';

export const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [ categories, setCategories ] = useState(['One Punch']);
    // const handleAdd = () => {
    //     // 'HunterXHunter'
    //     // categories.push('HunterXHunter');
    //     // console.log(categories);
    //     setCategories( ['HunterXHunter', ...categories ] );
    //     setCategories( cats => [  ...categories, 'HunterXHunter' ] );
    //     // console.log(categories);
    // }

    // console.log(categories);

    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories = { setCategories }/>
            {/* <AddCategory /> */}
            <hr />
{/* 
            <button
                onClick={ 
                    handleAdd
                }
            >
            Agregar
            </button> */}

            {/* <ol>
                <li>One</li>
                <li>Samurai</li>
                <li>Dragon</li>
            </ol> */}

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key = { category } 
                            category = { category } 
                        />
                        // <li key = { category }> { category }</li>
                    ))
                }
            </ol>

        </>
    );
}

export default GifExpertApp;

