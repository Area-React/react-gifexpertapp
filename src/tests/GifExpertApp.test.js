import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Prueba para GifExpertApp', () => {

    let wrapper; //undefined
    // let wrapper = shallow( <CounterApp /> ); 
    beforeEach( () => {
       wrapper = shallow(
           <GifExpertApp /> 
       );
        // console.log('beforeEach!!!')
    });

    test('Debe mostrar el texto <GifExpertApp /> ', () => {
        const text = 'GifExpertApp';
        console.log(text);
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar la linea <hr /> ', () => {
        // const linea = '';
        // console.log(linea);
        const textoValor = wrapper.find('hr').text().trim();
        expect( textoValor ).toMatchSnapshot();
    });

    test('Debe mostrar el botÃ³n Agregar ', () => {
      const boton = wrapper.find('button').at(0).simulate('click');
      console.log(boton.html());
    });
});
