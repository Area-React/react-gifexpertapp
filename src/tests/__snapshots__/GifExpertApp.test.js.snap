// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para GifExpertApp Debe mostrar el texto <GifExpertApp />  1`] = `ShallowWrapper {}`;

exports[`Prueba para GifExpertApp Debe mostrar la linea <hr />  1`] = `""`;
